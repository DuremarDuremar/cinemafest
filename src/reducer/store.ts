import { applyMiddleware, createStore, combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

import { responsReducer } from "./reducers/responsR";
import { menuReducer } from "./reducers/menuR";
import { filmsReducer } from "./reducers/filmsR";
import { transitionReducer } from "./reducers/transitionR";

const rootReducer = combineReducers({
  respons: responsReducer,
  menu: menuReducer,
  films: filmsReducer,
  transition: transitionReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);
