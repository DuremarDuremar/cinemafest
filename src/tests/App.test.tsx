import { render, screen } from "@testing-library/react";
import Header from "../components/fest_header";
import { Provider } from "react-redux";

import { store } from "../reducer/store";

const renderWithRedux = (component: any, {} = {}) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
};

const setSelectedProperty = jest.fn();

describe("Header", () => {
  it("header render", () => {
    renderWithRedux(
      <Header choiceFest={"Cannes"} setChoiceFest={setSelectedProperty} />
    );

    expect(screen.getByText("Cinema Festivals")).toBeInTheDocument();
  });
  it("header snapshot", () => {
    const component = renderWithRedux(
      <Header choiceFest={"Cannes"} setChoiceFest={setSelectedProperty} />
    );
    expect(component).toMatchSnapshot();
  });
});
