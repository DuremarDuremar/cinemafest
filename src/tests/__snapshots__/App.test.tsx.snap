// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM kDvdYm"
      >
        <section
          class="sc-gtsrHT jnEPae"
        >
          <ul>
            <li
              class="sc-dlnjwi kBDWEO"
            >
              Cannes
            </li>
            <li
              class="sc-dlnjwi hgxtDD"
            >
              Venice
            </li>
            <li
              class="sc-dlnjwi hgxtDD"
            >
              Berlin
            </li>
            <li
              class="sc-dlnjwi hgxtDD"
            >
              Sundance
            </li>
          </ul>
        </section>
        <section
          class="sc-hKFxyN dUEzus"
        >
          <div>
            <img
              alt="logo"
              src="logo.png"
            />
            <p>
              Cinema Festivals
            </p>
          </div>
          <div />
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM kDvdYm"
    >
      <section
        class="sc-gtsrHT jnEPae"
      >
        <ul>
          <li
            class="sc-dlnjwi kBDWEO"
          >
            Cannes
          </li>
          <li
            class="sc-dlnjwi hgxtDD"
          >
            Venice
          </li>
          <li
            class="sc-dlnjwi hgxtDD"
          >
            Berlin
          </li>
          <li
            class="sc-dlnjwi hgxtDD"
          >
            Sundance
          </li>
        </ul>
      </section>
      <section
        class="sc-hKFxyN dUEzus"
      >
        <div>
          <img
            alt="logo"
            src="logo.png"
          />
          <p>
            Cinema Festivals
          </p>
        </div>
        <div />
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
